packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -70,52 +74,82 @@ export default function TemplateSwitcher( {
 		setThemePreviewVisible( () => false );
 	};
 
+	const [ homeId, setHomeId ] = useState();
+
+	useEffect( () => {
+		const effect = async () => {
+			try {
+				const { success, data } = await fetch(
+					addQueryArgs( '/', { '_wp-find-template': true } )
+				).then( ( res ) => res.json() );
+				if ( success ) {
+					let newHomeId = data.ID;
+					if ( newHomeId === null ) {
+						const { getEntityRecords } = resolveSelect( 'core' );
+						newHomeId = (
+							await getEntityRecords( 'postType', 'wp_template', {
+								resolved: true,
+								slug: data.post_name,
+							} )
+						 )[ 0 ].id;
+					}
+					setHomeId( newHomeId );
+				} else {
+					throw new Error();
+				}
+			} catch ( err ) {
+				setHomeId( null );
+			}
+		};
+		effect();
```

Comment:
There is now a `findTemplate` util in `edit-site/src/utils`.

Reply:
59d0688244fd4b51ead82c889faf07e68c992db6